File: src/main/java/axirassa/model/PingerFrequency.java

package axirassa.model;

import axirassa.webapp.utilities.LabeledObject;

public enum PingerFrequency implements LabeledObject {
	MINUTE(60),
	MINUTES_5(5 * 60),
	MINUTES_10(10 * 60),
	MINUTES_15(15 * 60),
	MINUTES_30(30 * 60),
	MINUTES_60(60 * 60);
	private int interval;


	PingerFrequency (int interval) {
		this.interval = interval;
	}


	/**
	 * @return the number of seconds in this interval
	 */
	public int getInterval () {
		return interval;
	}


	@Override
	public String getLabel () {
		if (interval == 1)
			return "second";
		else if (interval < 60)
			return "" + interval + " seconds";
		else if (interval == 60)
			return "minute";
		else
			return "" + (interval / 60) + " minutes";
	}
}

File: src/main/java/axirassa/model/PingerFrequency.scala
package axirassa.model

class PingerFrequency(interval : Int) {
  def getInterval = interval

  def getLabel {
    interval match {
      case 1            => "second"
      case e if e < 60  => interval+"seconds"
      case e if e == 60 => "minute"
      case e            => (interval / 60)+" minutes"
    }
  }
}

object PingerFrequency {
  val MINUTE = new PingerFrequency(5)
  val MINUTES_5 = new PingerFrequency(5 * 60)
  val MINUTES_10 = new PingerFrequency(10 * 60)
  val MINUTES_15 = new PingerFrequency(15 * 60)
  val MINUTES_30 = new PingerFrequency(30 * 60)
  val MINUTES_60 = new PingerFrequency(60 * 60)
}